src/credentials/sso/ssoAccessTokenProvider.ts

```diff
@@ -4,93 +4,103 @@
  */
 
 import * as vscode from 'vscode'
-import { ClientRegistration } from './clientRegistration'
-import { AccessToken } from './accessToken'
+import { SSOOIDC } from 'aws-sdk'
+import { SsoClientRegistration } from './ssoClientRegistration'
+import { SsoAccessToken } from './ssoAccessToken'
 import { DiskCache } from './diskCache'
-import { Authorization } from './authorization'
 import { getLogger } from '../../shared/logger'
+import { StartDeviceAuthorizationResponse } from 'aws-sdk/clients/ssooidc'
 
 const CLIENT_REGISTRATION_TYPE = 'public'
+const CLIENT_NAME = `aws-toolkit-vscode-${Date.now()}`
+// According to Spec 'SSO Login Token Flow' the grant type must be the following string
 const GRANT_TYPE = 'urn:ietf:params:oauth:grant-type:device_code'
+// Used to convert seconds to milliseconds
+const MILLISECONDS_PER_SECOND = 1000
+const BACKOFF_TIME = 5000
 
 export class SsoAccessTokenProvider {
     private ssoRegion: string
     private ssoUrl: string
-    private ssoOidcClient: AWS.SSOOIDC
+    private ssoOidcClient: SSOOIDC
     private cache: DiskCache
 
-    constructor(ssoRegion: string, ssoUrl: string, ssoOidcClient: AWS.SSOOIDC, cache: DiskCache) {
+    constructor(ssoRegion: string, ssoUrl: string, ssoOidcClient: SSOOIDC, cache: DiskCache) {
         this.ssoRegion = ssoRegion
         this.ssoOidcClient = ssoOidcClient
         this.cache = cache
         this.ssoUrl = ssoUrl
     }
 
-    public async accessToken(): Promise<AccessToken> {
+    public async accessToken(): Promise<SsoAccessToken> {
         const accessToken = this.cache.loadAccessToken(this.ssoUrl)
         if (accessToken) {
             return accessToken
         }
+        const registration = await this.registerClient()
+        const authorization = await this.authorizeClient(registration)
+        const token = await this.pollForToken(registration, authorization)
+        this.cache.saveAccessToken(this.ssoUrl, token)
+        return token
+    }
 
-        try {
-            const token = await this.pollForToken()
-            this.cache.saveAccessToken(this.ssoUrl, token)
-            return token
-        } catch (err) {
-            throw err
-        }
+    public invalidate() {
+        this.cache.invalidateAccessToken(this.ssoUrl)
     }
 
-    private async pollForToken(): Promise<AccessToken> {
-        const registration = await this.registerClient()
-        const authorization = await this.authorizeClient(registration)
-        const deviceCodeExpiration = Date.now() + authorization.expiresIn * 1000
+    private async pollForToken(
+        registration: SsoClientRegistration,
+        authorization: StartDeviceAuthorizationResponse
+    ): Promise<SsoAccessToken> {
+        // Calculate the device code expirtation in milliseconds
+        const deviceCodeExpiration = Date.now() + authorization.expiresIn! * MILLISECONDS_PER_SECOND
```

Comment:
There's an opportunity for a local utility method that could serve this and line 80